app/actions/remote/scheduled_post.test.ts

```diff
@@ -0,0 +1,125 @@
+        },
+    ],
+};
+
+const throwFunc = () => {
+    throw Error('error');
+};
+
+const mockClient = {
+    getScheduledPostsForTeam: jest.fn(() => Promise.resolve({...scheduledPostsResponse})),
+    deleteScheduledPost: jest.fn((scheduledPostId) => {
+        return Promise.resolve(scheduledPostsResponse.bar.find((post) => post.id === scheduledPostId));
+    }),
+};
+jest.mock('@queries/servers/system', () => ({
+    getConfigValue: jest.fn(),
+}));
+
+const mockedGetConfigValue = jest.mocked(getConfigValue);
+let operator: ServerDataOperator;
+
+beforeAll(() => {
+    // eslint-disable-next-line
+    // @ts-ignore
+    NetworkManager.getClient = () => mockClient;
+});
+
+beforeEach(async () => {
+    await DatabaseManager.init([serverUrl]);
+    operator = DatabaseManager.getServerDatabaseAndOperator(serverUrl).operator;
+});
+
+afterEach(async () => {
+    await DatabaseManager.destroyServerDatabase(serverUrl);
+});
+
+describe('fetchScheduledPosts', () => {
+    it('fetch Schedule post - handle database not found', async () => {
+        const result = await fetchScheduledPosts('foo', 'bar');
+        expect(result).toBeDefined();
+        expect(result.error).toBeTruthy();
+    });
+
+    it('fetch Schedule post - handle client error', async () => {
+        jest.spyOn(NetworkManager, 'getClient').mockImplementationOnce(throwFunc);
+
+        const result = await fetchScheduledPosts(serverUrl, 'bar');
+        expect(result).toBeDefined();
+        expect(result.error).toBeTruthy();
+    });
+
+    it('fetch Schedule post - handle scheduled post disabled', async () => {
+        const result = await fetchScheduledPosts(serverUrl, 'bar');
+        expect(result).toBeDefined();
+        expect(result.scheduledPosts).toEqual([]);
+    });
+
+    it('fetch Schedule post - handle scheduled post enabled', async () => {
+        mockedGetConfigValue.mockResolvedValueOnce('true');
+        jest.spyOn(operator, 'handleScheduledPosts').mockResolvedValueOnce([]);
+        const result = await fetchScheduledPosts(serverUrl, 'bar');
+        expect(result).toBeDefined();
+        expect(result.scheduledPosts).toEqual(scheduledPostsResponse.bar);
+    });
+});
+
+describe('deleteScheduledPost', () => {
+    it('delete Schedule post - handle database not found', async () => {
+        const result = await deleteScheduledPost('foo', 'scheduled_post_id');
+        expect(result).toBeDefined();
+        expect(result.error).toBeTruthy();
+    });
+
+    it('delete Schedule post - handle client error', async () => {
+        jest.spyOn(NetworkManager, 'getClient').mockImplementationOnce(throwFunc);
+
+        const result = await deleteScheduledPost(serverUrl, 'scheduled_post_id');
+        expect(result).toBeDefined();
+        expect(result.error).toBeTruthy();
+    });
+
+    it('delete Schedule post - handle scheduled post enabled', async () => {
+        jest.spyOn(operator, 'handleScheduledPosts').mockResolvedValueOnce([]);
```

Comment:
Same as above, should we mock or check that it was calledWith() properly?

Reply:
Done.